tests/simple/testdata/dynamic.textproto

```diff
@@ -1,2 +1,2161 @@
+  }
+}
+section {
+  name: "uint64"
+  description: "Tests for uint64 conversion."
+  test {
+    name: "literal"
+    expr: "google.protobuf.UInt64Value{value: 123u}"
+    value { uint64_value: 123 }
+  }
+  test {
+    name: "literal_no_field_access"
+    expr: "google.protobuf.UInt64Value{value: 123u}.value"
+    disable_check: true
+    eval_error {
+      errors { message: "no_matching_overload" }
+    }
+  }
+  test {
+    name: "literal_zero"
+    expr: "google.protobuf.UInt64Value{}"
+    value { uint64_value: 0 }
+  }
+  test {
+    name: "var"
+    expr: "x"
+    type_env {
+      name: "x"
+      ident { type { message_type: "google.protobuf.UInt64Value" } }
+    }
+    bindings {
+      key: "x"
+      value { value { object_value {
+            [type.googleapis.com/google.protobuf.UInt64Value]
+            { value: 2000000 }
+          } } }
+    }
+    value { uint64_value: 2000000 }
+  }
+  test {
+    name: "field_assign_proto2"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint64_wrapper: 432u}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_uint64_wrapper { value: 432 } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto2_zero"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint64_wrapper: 0u}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_uint64_wrapper { } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto3"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_uint64_wrapper: 975u}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto3.TestAllTypes]
+        { single_uint64_wrapper { value: 975 } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto3_zero"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_uint64_wrapper: 0u}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto3.TestAllTypes]
+        { single_uint64_wrapper { } }
+      }
+    }
+  }
+  test {
+    name: "field_read_proto2"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint64_wrapper: 5123123123u}.single_uint64_wrapper"
+    value { uint64_value: 5123123123 }
+  }
+  test {
+    name: "field_read_proto2_zero"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint64_wrapper: 0u}.single_uint64_wrapper"
+    value { uint64_value: 0 }
+  }
+  test {
+    name: "field_read_proto2_unset"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{}.single_uint64_wrapper"
+    value { null_value: NULL_VALUE }
+  }
+}
+section {
+  name: "float"
+  description: "Tests for float conversion."
+  test {
+    name: "literal"
+    expr: "google.protobuf.FloatValue{value: -1.5e3}"
```

Comment:
I suspect if you add a test for the following the conversion will look problematic due to the higher precision of the double:

```
expr: "google.protobuf.FloatValue{value: 1.333e3}"
value: { double_value: ??? }
```

... but, if you were to test round-tripping you'd get what you want:

```
expr: "google.protobuf.FloatValue{value: google.protobuf.DoubleValue{value: 1.333e3}}"
value: { double_value: 1333.0 }
```

Reply:
That actually works fine (just checked it), since 1.333e3 has an exact representation as a float.  It fails for plain 1.333 however:

```
Eval got [double_value: 1.3329999446868896
            ], want [double_value: 1.333
            ]
```

I'll add a test to confirm that we're actually bottlenecking the precision to a float at some point.

Reply:
Excellent, I'm glad you had this on your radar. Perhaps just file it as an Issue so we don't forget about it? If you haven't already that is.

Reply:
Already added the test.